prog/weave-npc/main.go

```diff
@@ -218,8 +218,17 @@ func ipsetExist(ips ipset.Interface, name ipset.Name) (bool, error) {
 	return false, nil
 }
 
+func stopOnPanicRecover(stopChan chan struct{}) {
+	if r := recover(); r != nil {
+		os.Exit(1)
```

Comment:
What's the user experience if this happens?  Can they understand where the real problem occurred?